go-backend/db/recognition_hi5.go

```diff
@@ -0,0 +1,78 @@
+package db
+
+import (
+	"context"
+	logger "github.com/sirupsen/logrus"
+	"time"
+)
+
+const (
+	createRecognitionHi5Query = `INSERT INTO recognition_hi5 (
+		recognition_id,
+		comment,
+		given_by,
+		given_at
+		) VALUES ($1, $2, $3, $4);`
+
+	updateHi5QuotaQuery = `UPDATE users SET hi5_quota_balance = $1 where id = $2;`
+)
+
+type RecognitionHi5 struct {
+	Id				int			`db:"id" json:"id"`
+	RecognitionId			int			`db:"recognition_id" json:"recognition_id"`
+	Comment				string			`db:"comment" json:"comment"`
+	GivenBy				int			`db:"given_by" json:"given_by"`
+	GivenAt				int64			`db:"given_at" json:"given_at"`
+}
+
+func (reqHi5 *RecognitionHi5) CheckHi5QuotaBalance(hi5Quota int)(errorResponse map[string]ErrorResponse, valid bool){
+	if hi5Quota <= 0 {
+		logger.Error("Not enough Hi5 quota balance")
+		errorResponse = map[string]ErrorResponse {
```

Comment:
the error response should be prepared by handler function. This method can simply return an error object using errors.New(). Also after that logging can be done by the invoker of the function

Right now the method signature is a little complicated as it returns a boolean and error response. I suggest only returning an error if the balance is not remaining